staging/src/k8s.io/client-go/applyconfigurations/flowcontrol/v1beta1/flowschemastatus.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1beta1
+
+// FlowSchemaStatusApplyConfiguration represents an declarative configuration of the FlowSchemaStatus type for use
+// with apply.
+type FlowSchemaStatusApplyConfiguration struct {
+	Conditions *[]FlowSchemaConditionApplyConfiguration `json:"conditions,omitempty"`
+}
+
+// FlowSchemaStatusApplyConfiguration constructs an declarative configuration of the FlowSchemaStatus type for use with
+// apply.
+func FlowSchemaStatus() *FlowSchemaStatusApplyConfiguration {
+	return &FlowSchemaStatusApplyConfiguration{}
+}
+
+// SetConditions sets the Conditions field in the declarative configuration to the given value.
+func (b *FlowSchemaStatusApplyConfiguration) SetConditions(value []FlowSchemaConditionApplyConfiguration) *FlowSchemaStatusApplyConfiguration {
```

Comment:
How does this look when used? is it easy to add things one at a time? Is there a missing List type that maybe we should generate?

Reply:
This needs work. 

Previously had been generating a List typeref for each type, which resulted in:

```
.SetConditions(flowcontrolv1beta1apply.FlowSchemaConditionList{
  flowcontrolv1beta1apply.FlowSchemaCondition().
    SetSomeField(),
})
```

This didn't make have that much of an advantage over just letting the caller create the list:

```go
.SetConditions([]flowcontrolv1beta1apply.FlowSchemaConditionApplyConfiguration{
  flowcontrolv1beta1apply.FlowSchemaCondition().
    SetSomeField(),
})
```

I considered adding some functions to the list typerefs, but hadn't found any that proved particularly helpful. A `Add<ListElement>() <ListElement>` function doesn't chain well, and while a `Add<ListElement>` function that takes a function argument fixes the chaining problem, it ends up being quite verbose:

```go
.SetConditions(flowcontrolv1beta1apply.FlowSchemaConditionList().
  AddCondition(func (condition flowcontrolv1beta1apply.FlowSchemaCondition) {
    condition.SetSomeField()
  }),
)
```

Any thoughts on how to make the list typerefs more useful are welcome. There are probably ways to use them I haven't thought of.

We are currently considering using varadic arguments, that would look like:

```go
.SetConditions(
  flowcontrolv1beta1apply.FlowSchemaCondition().
    SetSomeField(),
)
```

Note that this works out pretty well for lists, but is not an option for maps.

Reply:
One idea maps- if we generated an "Entry" type for each type, we could use varargs for maps too, e.g.:

```go
.SetMapField(appsv1apply.MapEntry(<key>, <value>), ...), 
)
```

Reply:
Maybe I haven't grasped the full problem, but from my understanding we're saying something along the lines of using varargs would be preferred over allowing the user to create the list themselves?

```go
.SetConditions(
  flowcontrolv1beta1apply.FlowSchemaCondition().
    SetSomeField(),
)
```
vs

```go
.SetConditions([]flowcontrolv1beta1apply.FlowSchemaConditionApplyConfiguration{
  flowcontrolv1beta1apply.FlowSchemaCondition().
    SetSomeField(),
})
```

The only difference seems to be declaring the list type vs not declaring it and have the code automatically convert the varargs to a list? I actually think explicitly declaring the list value types has some benefits. For example, we only generate applyconfigs for structs that have json tagged fields.

Eg: `[]*flowcontrolv1beta.FlowSchemaCondition` is generated as `[]*flowcontrolv1beta.FlowSchemaConditionApplyConfiguration` while `[]*metav1.Time` would be unchanged `[]*metav1.Time`.

Even without the type declaration, users would still have to figure out the proper type to use which probably already involves a bit of digging into the code. Explicitly declaring the types doesn't seem to add that much extra work.

Reply:
One thing I've been trying to do it not force the developer to have to directly reference the full *ApplyConfiguration types.  One thing I like about the [passing integration test](https://github.com/kubernetes/kubernetes/pull/99214/commits/e57168d2f3b95246c30d80f220e456c814d93342#diff-7995de47f252f7dd3bf8a2fbef7390af61f512970ccc531546e9520270f9cabbR811) right now is that you don't see "ApplyConfiguration" anywhere.  The varargs do help with that.  I'll admit, this is a very subject preference I have, but it looks nice to me that way.  I do agree it's not strictly needed.

On a unrelated note, I fixed the generated code to always take pointers to apply configuration types since this is what the constructors return, I had to jump through some [hoops](https://github.com/kubernetes/kubernetes/pull/99012/commits/8887cfb192ff97a35be1c3e83f842424f2c75640#diff-d1da593cc5a517f36653eca44bc9f8c815fb7c144daa45238d338e82c3140d23R61) to support slices, but it seems worth it.   We don't need to worry about maps much since there are no maps in k/k that have apply configuration values (since associative lists are always used); The generator will produce a working setter if anyone ever uses a type like that, it just won't be as convenient.

Reply:
Thanks for the explanation! This does make the builder much cleaner and it looks like the additional code isn't too complex. +1